events/system/interactionCreate.js

```diff
@@ -29,6 +30,24 @@ export default {
         });
       }
 
+      if (command.guards?.length) {
+        for (const guardName of command.guards) {
+          const guard = guardMap[guardName];
+
+          if (!guard) {
+            console.error(`Unknown guard: ${guardName}`);
+            await interaction.reply({
+              content: "Configuration error.",
+              flags: MessageFlags.Ephemeral,
+            });
+            return;
+          }
+
+          const passed = await guard(interaction);
+          if (!passed) return;
```

Comment:
Shouldn't we inform the user of their failure? Or we just going to leave them hangin?